src/uct/ib/rc/base/rc_iface.h

```diff
@@ -361,13 +390,9 @@ uct_rc_iface_tag_imm_data_pack(uint32_t *ib_imm, uint32_t *app_ctx,
 }
 
 static UCS_F_ALWAYS_INLINE uint64_t
-uct_rc_iface_tag_imm_data_unpack(struct ibv_exp_wc *wc, uint32_t app_ctx)
+uct_rc_iface_tag_imm_data_unpack(uint32_t ib_imm, uint32_t app_ctx, int is_imm)
 {
-    if (wc->exp_wc_flags & IBV_EXP_WC_WITH_IMM) {
-        return ((uint64_t)app_ctx << 32) | wc->imm_data;
-    } else {
-        return 0ul;
-    }
+    return is_imm ? ((uint64_t)app_ctx << 32) | ib_imm : 0ul;
```

Comment:
maybe add more ( ) here around what is between ? and :

Reply:
ok